src/scripts/reducers/UserReducer.js

```diff
@@ -0,0 +1,37 @@
+/* @flow */
+
+import { handleActions } from 'redux-actions'
+
+import type { Action } from 'types/ApplicationTypes'
+import { LOGIN_REQUESTED, LOGIN_SUCCESS, LOGOUT } from 'constants/ActionConstants'
+import UserRecord from 'records/UserRecords'
+
+const reducer = {
+  [LOGIN_REQUESTED]: (
+    state: UserRecord
+  ): UserRecord => {
+    return state
+      .set('isLoggingIn', true)
+      .set('email', null)
+      .set('name', null)
+  },
```

Comment:
☝️  the logic here is fine, but it is more performant to do this in one operation:

```javascript
return state.merge({
  isLoggingIn: true,
  email: null,
  name: null
});
```